conformance/submission_details_template.txt

```diff
@@ -58,10 +58,12 @@ Target SYCL device:
 # and device combinations.
 #
 SYCL implementation version(s):
+Device(s):
+Platform(s):
```

Comment:
Can we remove "Platform" here, and also above under "Conformance submission details"?  This is just redundant information because it's the same as "Backend".

Reply:
WG proposes that we could combine these two lines as:

> Backend / platform(s):